src/client/Microsoft.Identity.Client/Kerberos/KerberosSupplementalTicketManager.cs

```diff
@@ -0,0 +1,236 @@
+            }
+
+            string kerberosAsRep = jwt.GetValueOrEmptyString(KerberosClaimType);
+            if (string.IsNullOrEmpty(kerberosAsRep))
+            {
+                return null;
+            }
+
+            return (KerberosSupplementalTicket)JsonConvert.DeserializeObject(kerberosAsRep, typeof(KerberosSupplementalTicket));
+        }
+
+        /// <summary>
+        /// Save current Kerberos Ticket to current user's Ticket Cache.
+        /// </summary>
+        /// <param name="ticket">Kerberos ticket object to save.</param>
+        /// <remarks>Can throws <see cref="ArgumentException"/> when given ticket parameter is not a valid Kerberos Supplemental Ticket.
+        /// Can throws <see cref="Win32Exception"/> if error occurs while saving ticket information into Ticket Cache.
+        /// </remarks>
+        public static void SaveToCache(KerberosSupplementalTicket ticket)
+        {
+            SaveToCache(ticket, DefaultLogonId);
+        }
+
+        /// <summary>
+         /// Save current Kerberos Ticket to current user's Ticket Cache.
+         /// </summary>
+         /// <param name="ticket">Kerberos ticket object to save.</param>
+         /// <param name="logonId">The Logon Id of the user owning the ticket cache.
+         /// The default of 0 represents the currently logged on user.</param>
+         /// <remarks>Can throws <see cref="ArgumentException"/> when given ticket parameter is not a valid Kerberos Supplemental Ticket.
+         /// Can throws <see cref="Win32Exception"/> if error occurs while saving ticket information into Ticket Cache.
+         /// </remarks>
+        public static void SaveToCache(KerberosSupplementalTicket ticket, long logonId)
+        {
+#if SUPPORT_KERBEROS
+            if (ticket == null || string.IsNullOrEmpty(ticket.KerberosMessageBuffer))
+            {
+                throw new ArgumentException("Kerberos Ticket information is not valid");
+            }
+
+            using (var cache = Win32.TicketCacheWriter.Connect())
+            {
+                byte[] krbCred = Convert.FromBase64String(ticket.KerberosMessageBuffer);
+                cache.ImportCredential(krbCred, logonId);
+            }
+#endif
+        }
+
+        /// <summary>
+        /// Checks a Kerberos Service Ticket associated with given service principal name exists
+        /// in current user's Ticket Cache.
+        /// </summary>
+        /// <param name="servicePrincipalName">Service principal name to find associated Kerberos Ticket.</param>
+        /// <returns>True if Kerberos Ticket exists. False, otherwise.</returns>
+        /// <returns>True if Ticket save to Ticket Cache successfully. False, otherwise.</returns>
+        /// <remarks>
+        /// Can throws <see cref="Win32Exception"/> if error occurs while searching ticket information from Ticket Cache.
+        /// </remarks>
+        public static bool IsTKerberosTicketExistsInCache(string servicePrincipalName)

```

Comment:
Is looks funny. I guess "Does" would be better, but I understand the conversion "Is" for methods returning bools ...

Reply:
This interface is not required - removed.